src/apps/gas-tank/hooks/useOffer.tsx

```diff
@@ -0,0 +1,498 @@
+       */
+      const fromAmountInWei = parseUnits(String(fromAmount), fromTokenDecimals);
+      const feeDeduction = fromAmountInWei / BigInt(100); // 1% fee
+      const fromAmountFeeDeducted = fromAmountInWei - feeDeduction;
+      const toTokenAddress = USDC_ADDRESSES[fromChainId];
+      const toTokenDecimals = 6; // USDC has 6 decimals
+      /**
+       * Step 3: Create route request for LiFi
+       * This request includes all necessary parameters for finding swap routes
+       */
+      const routesRequest: RoutesRequest = {
+        fromChainId,
+        toChainId: fromChainId, // Swapping within the same chain
+        fromTokenAddress: fromTokenAddressWithWrappedCheck,
+        toTokenAddress,
+        fromAmount: fromAmountFeeDeducted.toString(),
+        options: {
+          slippage,
+          bridges: {
+            allow: ['relay'],
+          },
+          exchanges: { allow: ['openocean', 'kyberswap'] },
+        },
+      };
+
+      const result = await getRoutes(routesRequest);
+      const { routes } = result;
+
+      const allOffers = routes as Route[];
+
+      if (allOffers.length) {
+        /**
+         * Step 4: Find the best offer
+         * Compare all available routes and select the one with the highest output
+         */
+        const bestOffer = allOffers.reduce((a, b) => {
+          const receiveAmountA = processEth(a.toAmount, toTokenDecimals);
+          const receiveAmountB = processEth(b.toAmount, toTokenDecimals);
+          return receiveAmountA > receiveAmountB ? a : b;
+        });
+
+        const selectedOffer: SwapOffer = {
+          tokenAmountToReceive: processEth(bestOffer.toAmount, toTokenDecimals),
+          offer: bestOffer as Route,
+        };
+
+        return selectedOffer;
+      }
+
+      // Return undefined instead of empty object when no routes found
+      return undefined;
+    } catch (e) {
+      console.error(
+        'Sorry, an error occurred while trying to fetch the best swap offer. Please try again.',
+        e
+      );
+      // Return undefined instead of empty object on error
+      return undefined;
+    }
+  };
+
+  /**
+   * Check if token allowance is set for a specific spender
+   * This function verifies if the wallet has approved a contract to spend tokens
+   */
+  const isAllowanceSet = async ({
+    owner,
+    spender,
+    tokenAddress,
+    chainId,
+  }: {
+    owner: string;
+    spender: string;
+    tokenAddress: string;
+    chainId: number;
+  }) => {
+    if (isZeroAddress(tokenAddress)) return undefined;
+
+    // Validate inputs
+    if (!owner || !spender || !tokenAddress) {
+      console.warn('Invalid inputs for allowance check:', {
+        owner,
+        spender,
+        tokenAddress,
+      });
+      return undefined;
+    }
+
+    try {
+      const publicClient = createPublicClient({
+        chain: getNetworkViem(chainId),
+        transport: http(),
+      });
```

Comment:
Whilst this is not wrong - do we have a client we can use already from something that's been instantiated?